micro-rdk/src/common/config_monitor.rs

```diff
@@ -72,15 +78,61 @@ where
                 .await?;
 
             if new_config
+                .as_ref()
                 .config
-                .is_some_and(|cfg| cfg != *self.curr_config)
+                .as_ref()
+                .is_some_and(|cfg| *cfg != *self.curr_config)
             {
+                log::info!("new robot config detected");
+
+                #[cfg(feature = "ota")]
+                {
+                    let config = new_config.config.clone().unwrap();
+                    if let Some(service) = config
+                        .services
+                        .iter()
+                        .find(|&service| service.model == *ota::OTA_MODEL_TRIPLET)
+                    {
+                        match ota::OtaService::from_config(
+                            service,
+                            self.storage.clone(),
+                            self.executor.clone(),
+                        ) {
+                            Ok(mut ota) => {
+                                let curr_metadata = ota.stored_metadata().await;
```

Comment:
I feel like the logic here could go back to the OTA service and this could be reduced to a function call on the service? You would no longer need the `stored_metadata` function and the `pending_version` would no longer need to be `pub(crate)`

Reply:
you're right. because this would run every 10 seconds when the config gets checked, I wanted to control how often logging gets propogated, otherwise we's see `version up to date` all the time.

I've since moved a single log to when viam server starts to just log the current firmware once, and report only the update target when attempting ota